constructShortestPath.js

```diff
@@ -0,0 +1,94 @@
+/*
+We can represent the input data as a graph where the cities are reprsented as nodes and the roads as edges. 
+Also by looking at the input, we can represent the graph as undirected and build an adjacency list.
+Thus, for finding the shortest path between two cities, we can find the shorest path the same way 
+we would between nodes in our graph using a BFS.
+
+But we need to find the path itself. So when we reach a node for the first time, the node previously
+would be the ideal node to come from since it will be the shorest path. Thus, be can build a object
```

Comment:
This sentence "So when we reach a node for the first time, the node previously would be the ideal node to come from since it will be the shorest path." is really the meat of your answer, and you need to do more to explain WHY this is true. It also should be stated more clearly; this sentence is pretty confusing.

Reply:
So the way we've written our BFS, we only process a node once per node and it will be the first time we've visited that node because of our visited set. And since we're using a BFS to find the actual path, you're guaranteed the shortest path. The current node will always have the ideal node before it which in-turn will give us an actual path from the start node to the end node.

Reply:
Sure, this has the right ideas. I'll provide my own description in case this helps.

We've structured our BFS such that we process each node at most one time. We only add a node to the queue the very first time that we see it. Because a BFS goes level-by-level, the first time we see a node, we have used the shortest path to get to that node. Whenever we are adding a neighbor to the queue, we store our current node as the "ideal previous node" for the neighbor. This means that in the shortest path from the start to the neighbor, our current node comes just before the neighbor.